### ISSUE TITLE
"host must not be null" exception in EC2Client, AutoScalingClient

### ISSUE THREAD

**Issue by lmittmann** [aws/aws-sdk-java-v2#401]
Hi, when I execute requests with the **EC2Client** or **AutoScalingClient** I get the NullPointerException: host must not be null.

```
Exception in thread "main" java.lang.NullPointerException: host must not be null.
	at software.amazon.awssdk.utils.Validate.paramNotNull(Validate.java:137)
	at software.amazon.awssdk.http.DefaultSdkHttpFullRequest.<init>(DefaultSdkHttpFullRequest.java:52)
	at software.amazon.awssdk.http.DefaultSdkHttpFullRequest.<init>(DefaultSdkHttpFullRequest.java:40)
	at software.amazon.awssdk.http.DefaultSdkHttpFullRequest$Builder.build(DefaultSdkHttpFullRequest.java:294)
	at software.amazon.awssdk.http.DefaultSdkHttpFullRequest$Builder.build(DefaultSdkHttpFullRequest.java:155)
	at software.amazon.awssdk.core.http.SdkHttpFullRequestAdapter.toHttpFullRequest(SdkHttpFullRequestAdapter.java:39)
	at software.amazon.awssdk.core.client.SyncClientHandlerImpl.execute(SyncClientHandlerImpl.java:95)
	at software.amazon.awssdk.core.client.SyncClientHandlerImpl.execute(SyncClientHandlerImpl.java:76)
	at software.amazon.awssdk.core.client.SdkClientHandler.execute(SdkClientHandler.java:45)
	at software.amazon.awssdk.services.autoscaling.DefaultAutoScalingClient.describeAutoScalingGroups(DefaultAutoScalingClient.java:1050)
	at last line of my code sniped below
```

## Expected Behavior
The host field in `DefaultSdkHttpFullRequest` should be set.

## Current Behavior
The host field in `DefaultSdkHttpFullRequest` is not set.

## Steps to Reproduce
This is the code I run. The exception is thrown by the last line.
```java
AwsCredentialsProvider awsCredentials = StaticCredentialsProvider.create(
		AwsCredentials.create(accessKey, secretKey));

AutoScalingClient asClient = AutoScalingClient.builder()
		.credentialsProvider(awsCredentials)
		.region(region)
		.build();

DescribeAutoScalingGroupsRequest request = DescribeAutoScalingGroupsRequest.builder()
		.autoScalingGroupNames(autoScalingGroupName)
		.build();
DescribeAutoScalingGroupsResponse response = asClient.describeAutoScalingGroups(request);
```

## Your Environment

* AWS Java SDK version used: `2.0.0-preview-7`
* JDK version used: `openjdk version "1.8.0_151"`
* Operating System and version: `ubuntu 16.04`

**Comment by millems** [comment]
Unfortunately I'm having issues reproducing the issue. What region are you specifying when you create the client?

**Comment by lmittmann** [comment]
I use the region us-east-1

**Comment by millems** [comment]
Hmm, this code works for me:

```Java
AutoScalingClient autoscaling = AutoScalingClient.builder()
                                                 .region(Region.US_EAST_1)
                                                 .build();
DescribeAutoScalingGroupsRequest request = DescribeAutoScalingGroupsRequest.builder()
                                                                           .autoScalingGroupNames(autoScalingGroupName)
                                                                           .build();
System.out.println(autoscaling.describeAutoScalingGroups(request));
```

Do you get any useful information with debug-logging enabled?

**Comment by lmittmann** [comment]
Hi, the cause for the issue was the region.

I deserialized the region form my projects config, however in the sdk v1 the region is an `enum` while in v2 it is an `AbstractEnum`. To deserialize the Region correctly I used the config string `US_EAST_1` in sdk v1 which has to be changed to `us-east-1` in sdk v2 apparently.

While this is not really a bug it is still a little inconvenient, since the string `US_EAST_1` is deserialized to a Region object without throwing any exception and also seems to be correct when being evaluated in debug mode.

Thanks for the help!
